netcontrold/tests/lib/test_data_coll.py

```diff
@@ -0,0 +1,287 @@
+
+        # create port class of name 'virtport1'.
+        dataif.make_dataif_port("port1")
+
+        # add port object into pmd.
+        self.port1 = self.fx_pmd_1.add_port("port1")
+        self.port1.numa_id = self.fx_pmd_1.numa_id
+
+        # add rxq to port object
+        self.rxq_1 = self.port1.add_rxq(0)
+
+        # create another pmd object.
+        self.fx_pmd_2 = dataif.Dataif_Pmd(13)
+
+        # let it be in numa 0.
+        self.fx_pmd_2.numa_id = 0
+
+        # add it to pmd_map
+        self.pmd_map[13] = self.fx_pmd_2
+
+        # create port class of name 'virtport2'.
+        dataif.make_dataif_port("port2")
+
+        # add port object into pmd.
+        self.port2 = self.fx_pmd_2.add_port("port2")
+        self.port2.numa_id = self.fx_pmd_2.numa_id
+
+        # add rxq to port object
+        self.rxq_2 = self.port2.add_rxq(0)
+
+        return
+
+    # Test case:
+    #   getting pmd stats from get_pmd_stats function and checking if
+    #   declared pmd_map is modified or not
+    @mock.patch('netcontrold.lib.util.exec_host_command', mock_pmd_stats)
+    def test_get_pmd_stats_1(self):
+        # create a copy of original pmd_map
+        expected = copy.deepcopy(self.pmd_map)
+
+        # copy both original pmd objects
+        expected_pmd_1 = expected[1]
+        expected_pmd_1.rx_cyc[expected_pmd_1.cyc_idx] = 1000
+        expected_pmd_1.idle_cpu_cyc[expected_pmd_1.cyc_idx] = 1100
+        expected_pmd_1.proc_cpu_cyc[expected_pmd_1.cyc_idx] = 1200
+
+        expected_pmd_2 = expected[13]
+        expected_pmd_2.rx_cyc[expected_pmd_2.cyc_idx] = 3000
+        expected_pmd_2.idle_cpu_cyc[expected_pmd_2.cyc_idx] = 3100
+        expected_pmd_2.proc_cpu_cyc[expected_pmd_2.cyc_idx] = 3200
+
+        # calling get_pmd_stats function. pmd_map is modified here
+        out = dataif.get_pmd_stats(self.pmd_map)
+
+        # copy both modified pmd objects
+        out_pmd_1 = out[1]
+        out_pmd_2 = out[13]
+
+        # check if original and modified pmd_map objects are different.
+        # if __eq__ returns false , then pmd objects are modified
+        self.assertEqual(out_pmd_1.__eq__(expected_pmd_1), True)
+        self.assertEqual(out_pmd_2.__eq__(expected_pmd_2), True)
+
+    # Test case:
+    #   getting interface stats from get_interface_stats function and checking if
+    #   declared port objects are modified or not
+    @mock.patch('netcontrold.lib.util.exec_host_command', mock_interface_stats)
+    def test_get_interface_stats_1(self):
+        # create a copy of original port1
+        expected = copy.deepcopy(self.port1)
+
+        # modify these to expected
+        expected.type = "dpdkvhostuserclient"
+        expected.tx_retry_cyc[expected.cyc_idx] = 0
+
+        # calling get_interface_stats function.port1 object is modified here
+        dataif.get_interface_stats()
+
+        # check if expected and modified port1 objects are same
+        self.assertEqual(self.port1.__eq__(expected), True)
```

Comment:
self.assertEqual(port1, expected_port1, "port1 to be matched")

Reply:
ok